\documentclass[11pt,]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1.0in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={NAME OF THIS STUDY},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\newcommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{\textbf{NAME OF THIS STUDY}}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{}
    \preauthor{}\postauthor{}
    \date{}
    \predate{}\postdate{}
  
\usepackage{helvet} % Helvetica font
\renewcommand*\familydefault{\sfdefault} % Use the sans serif version of the font
\usepackage[T1]{fontenc}

\usepackage[none]{hyphenat}

\usepackage{setspace}
\doublespacing
\setlength{\parskip}{1em}

\usepackage{lineno}

\usepackage{pdfpages}

\begin{document}
\maketitle

\begin{verbatim}
## Loading required package: foreach
\end{verbatim}

\begin{verbatim}
## Loading required package: iterators
\end{verbatim}

\begin{verbatim}
## Loading required package: parallel
\end{verbatim}

\begin{verbatim}
## Type 'citation("pROC")' for a citation.
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'pROC'
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:stats':
## 
##     cov, smooth, var
\end{verbatim}

\begin{verbatim}
## Loading required package: lattice
\end{verbatim}

\begin{verbatim}
## Loading required package: ggplot2
\end{verbatim}

\begin{verbatim}
## Loading required package: permute
\end{verbatim}

\begin{verbatim}
## This is vegan 2.5-2
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'vegan'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:caret':
## 
##     tolerance
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'gtools'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:permute':
## 
##     permute
\end{verbatim}

\begin{verbatim}
## -- Attaching packages ---------------------------------- tidyverse 1.2.1 --
\end{verbatim}

\begin{verbatim}
## v tibble  1.4.2     v purrr   0.2.5
## v tidyr   0.8.1     v dplyr   0.7.6
## v readr   1.1.1     v stringr 1.3.1
## v tibble  1.4.2     v forcats 0.3.0
\end{verbatim}

\begin{verbatim}
## -- Conflicts ------------------------------------- tidyverse_conflicts() --
## x purrr::accumulate() masks foreach::accumulate()
## x dplyr::filter()     masks stats::filter()
## x dplyr::lag()        masks stats::lag()
## x purrr::lift()       masks caret::lift()
## x purrr::when()       masks foreach::when()
\end{verbatim}

\begin{verbatim}
## pdf 
##   2
\end{verbatim}

\vspace{35mm}

Running title: INSERT RUNNING TITLE HERE

\vspace{35mm}

Begüm D. Topçuoğlu\^{}1, Jenna Wiens\^{}2, Patrick D.
Schloss\textsuperscript{1\(\dagger\)}

\vspace{40mm}

\(\dagger\) To whom correspondence should be addressed:
\href{mailto:pschloss@umich.edu}{\nolinkurl{pschloss@umich.edu}}

1. Department of Microbiology and Immunology, University of Michigan,
Ann Arbor, MI 48109

2. Department of Computer Science and Engineering, University or
Michigan, Ann Arbor, MI 49109

\newpage

\linenumbers

\subsection{Abstract}\label{abstract}

\newpage

\subsection{Introduction}\label{introduction}

As gut microbiome field continues to grow, there will be an
ever-increasing demand for reproducible machine learning methods to
analyze microbiome sequence read count data and to determine association
with a continuous or categorical phenotype of interest.

Colorectal cancer is one of the leading cause of death among cancers in
the United States. Early diagnosis increases the chance of survival.
However the current diagnostic methods are expensive and invasive. As a
less invasive tool, numerous studies use relative abundances of the gut
bacteria populations to predict disease progression. Most microbial
communities are pretty patchy and the likelihood of a single feature
that explains the differences in health is pretty small. It is likely
that many biomarkers are needed to account for the patchiness as well as
the context dependency of the features.

ML use in microbiome literature is a bit like the wild west with lack of
clarity over methods, testing, validation, etc. There is a need for
guidance on how to properly implement these different methods. We need
to emphasize good machine learning practices and pipelines and discuss
the reproducibility, robustness and actionability of models.

We established a non-leaky pipeline. We performed L1 and L2-regularized
logistic regression, Linear SVM, Non-Linear SVM, Decision tree, Random
forest, XGBoost and Feed Forward Neural Net classification models. We
evaluated the classification performance of different machine learning
methods. We also want to discuss the reproducibility, robustness,
actionability, interpretibility and susceptibility to overfitting of
each method.

Generalisation Perfomance of each model. Is there a maximum threshold of
prediction with all these methods? Does an increase in model complexity
improve predictibility? Synthesis statement regarding modeling 16S
microbiome data

\subsection{Results and Discussion}\label{results-and-discussion}

\subsection{Conclusions}\label{conclusions}

\subsection{Materials and Methods}\label{materials-and-methods}

\newpage

Insert figure legends with the first sentence in bold, for example:

\textbf{Figure 1. Number of OTUs sampled among bacterial and archaeal
16S rRNA gene sequences for different OTU definitions and level of
sequencing effort.} Rarefaction curves for different OTU definitions of
Bacteria (A) and Archaea (B). Rarefaction curves for the coarse
environments in Table 1 for Bacteria (C) and Archaea (D).

\newpage

\subsection{References}\label{references}


\end{document}
