\documentclass[11pt,]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1.0in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={Evalution of binary classification pipelines and methods for 16S rRNA gene data},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\newcommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{\textbf{Evalution of binary classification pipelines and methods for 16S
rRNA gene data}}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{}
    \preauthor{}\postauthor{}
    \date{}
    \predate{}\postdate{}
  
\usepackage{booktabs}
\usepackage{longtable}
\usepackage{array}
\usepackage{multirow}
\usepackage[table]{xcolor}
\usepackage{wrapfig}
\usepackage{float}
\usepackage{colortbl}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage[normalem]{ulem}
\usepackage{makecell}

\usepackage{helvet} % Helvetica font
\renewcommand*\familydefault{\sfdefault} % Use the sans serif version of the font
\usepackage[T1]{fontenc}

\usepackage[none]{hyphenat}

\usepackage{setspace}
\doublespacing
\setlength{\parskip}{1em}

\usepackage{lineno}

\usepackage{pdfpages}

\begin{document}
\maketitle

\vspace{35mm}

Running title: Machine learning methods in microbiome studies

\vspace{35mm}

Begüm D. Topçuoğlu\^{}1, Jenna Wiens\^{}2, Patrick D.
Schloss\textsuperscript{1\(\dagger\)}

\vspace{40mm}

\(\dagger\) To whom correspondence should be addressed:
\href{mailto:pschloss@umich.edu}{\nolinkurl{pschloss@umich.edu}}

1. Department of Microbiology and Immunology, University of Michigan,
Ann Arbor, MI 48109

2. Department of Computer Science and Engineering, University or
Michigan, Ann Arbor, MI 49109

\newpage

\linenumbers

\subsection{Abstract}\label{abstract}

\newpage

\subsection{Introduction}\label{introduction}

As gut microbiome field continues to grow, there will be an
ever-increasing demand for reproducible machine learning methods to
analyze 16S rRNA gene sequence data and to determine association of the
microbiome with a continuous or categorical phenotype of interest. The
use of machine learning in microbiome literature lack clarity over the
learning pipeline which spans the problem formulation, feature
selection, feature pre-processing, model learning, and output. There is
a need for guidance on how to properly implement good machine learning
practices to generate reproducible, robust and actionable models. We
also need to generate interpretable models for biomedical researchers to
adopt and use regularly. In this study we chose to focus on a binary
classification model that predicts if an individual has screen-relevant
tumors in their colon or not, to establish a model pipeline.

Colorectal cancer is one of the leading cause of death among cancers in
the United States. Each person in the industrialized world has on
average a one-in-twenty chance of developing colorectal cancer (CRC) in
their lifetime and once diagnosed, more than one-third will not survive
5 years. Colonoscopy as a screening tool is very effective, however it
is very invasive, expensive and have a low rate of patient adherence.
Therefore, there is a need for improved non-invasive methods to screen
individuals. One proposed non-invasive screening tool is using gut
microbiome-based biomarkers. Patients with colorectal cancer have
different stool community of microbes compared to adults with normal
colons. This difference however cannot be explained by a single or a
handful of features in the gut microbiome but by many of them in
relation to one another. Therefore, machine learning is a great tool to
investigate the differences between the gut microbiomes of CRC patients
and healthy individuals. Previous studies have shown that human
hemoglobin levels and bacterial population abundances in the stool help
us predict screen relevant growth in the colon, however the literature
for the problem of classification colorectal cancer diagnosis vary
greatly, with areas under the receiver operating characteristic curve
(AUC) of 0.7-0.8. The variation in classification performance is based
in part on differences in the task definition, in part on differences in
the study populations, and in part on the evaluation method. The highest
reported AUCs were from studies of \ldots{} Additionally, some of the
reported results were not obtained from testing on held-out sets. In
this study, we have defined classification pipelines with L2-regularized
logistic regression, L1 and L2 linear suppor vector machines (SVM),
radial basis function SVM , decision tree, random forest and XGBoost. We
evaluated the generalization and prediction performance of these
methods. We also compared each method based on their reproducibility,
robustness, actionability, interpretibility and susceptibility to
overfitting.

Generalisation Perfomance of each model. Is there a maximum threshold of
prediction with all these methods? Does an increase in model complexity
improve predictibility? Synthesis statement regarding modeling 16S
microbiome data

\subsection{Results and Discussion}\label{results-and-discussion}

\subsection{Conclusions}\label{conclusions}

\subsection{Materials and Methods}\label{materials-and-methods}

\paragraph{The data}\label{the-data}

We obtained stool OTU abundance data and metadata from the Sze et al.
(1). The stool OTU abundance data and metadata comes from the Great
Lakes- New England Early Detection Research Network study which
collected stool samples from eligible individuals. Briefly, eligible
patients for this study were aged at least 18 years and willing to
collect stool samples. Colonoscopies were performed and fecal samples
were collected from participants in four locations: Toronto (ON,
Canada), Boston (MA, USA), Houston (TX, USA), and Ann Arbor (MI, USA).
Patient diagnoses were determined by colonoscopic examination and
histopathological review of any biopsies taken. Patients with an adenoma
greater than 1 cm, more than three adenomas of any size, or an adenoma
with villous histology were classified as advanced adenoma. Fecal
material was used for Fecal Immunological Tests and these tests were
used to measure human hemoglobin concentrations. This study had 172
control, 198 adenomas and 120 carcinomas. Of the 198 adenomas, 109 were
advanced adenomas.

\newpage

\textbf{Figure 1. Generalization and classification performance of
modeling methods } AUC values of all cross validation and testing
performances. The boxplot shows quartiles at the box ends and the
statistical median as athe horizontal line in the box. The whiskers show
the farthest points that are not outliers. Outliers are data points that
are not within 3/2 times the interquartile ranges.


\end{document}
