\documentclass[11pt,]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1.0in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={Machine Learning Manuscript Outline},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\newcommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{Machine Learning Manuscript Outline}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{}
    \preauthor{}\postauthor{}
    \date{}
    \predate{}\postdate{}
  
\usepackage{booktabs}
\usepackage{longtable}
\usepackage{array}
\usepackage{multirow}
\usepackage[table]{xcolor}
\usepackage{wrapfig}
\usepackage{float}
\usepackage{colortbl}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage[normalem]{ulem}
\usepackage{makecell}

\usepackage{helvet} % Helvetica font
\renewcommand*\familydefault{\sfdefault} % Use the sans serif version of the font
\usepackage[T1]{fontenc}

\usepackage[none]{hyphenat}

\usepackage{setspace}
\doublespacing
\setlength{\parskip}{1em}

\usepackage{lineno}

\usepackage{pdfpages}
\floatplacement{figure}{H} % Keep the figure up top of the page

\begin{document}
\maketitle

\subsection{Introduction}\label{introduction}

\subparagraph{General Context of the
work}\label{general-context-of-the-work}

Advances in sequencing technology and decreasing costs of generating 16S
rRNA gene sequences allowed rapid exploration and taxonomic
characterization of the human associated microbiome. Currently, the
microbiome field is growing at an unprecedented rate and as a result,
there is an ever-increasing demand for reproducible methods for
identifying associations between members of the microbiome and human
health. Human associated microbial communities are remarkably uneven and
it is unlikely that a single species can explain a disease state
comprehensively. It is more likely that subsets of those communities, in
relation to one another and to their host, account for the unevenness
and can be used as biomarkers of a specific disease state. Thus,
researchers have started to explore the utility of machine learning (ML)
techniques to identify the biomarkers associated with healthy and
diseased individuals. However, currently the field's use of ML lacks
clarity and consistency on which methods are used, how these methods are
implemented and if these methods are reproducible. Moreover, there is a
lack of consideration on why a particular method is utilized. Often,
there is a trade-off between the interpretibility and accuracy of ML
methods. As researchers, we have to find the right balance to accomplish
complex knowledge tasks such as identifying a disease state based on
microbiome-associated biomarkers while also understanding how these
tasks are accomplished.

\subparagraph{Narrower research area and statement of its
importance}\label{narrower-research-area-and-statement-of-its-importance}

One application of machine learning to microbiome data has been to
classify patients as having colorectal tumors based on
microbiota-associated biomarkers. However, the efforts to combine ML
methods with relative abundances of bacterial poplations in stool to
predict CRC tumors have been negatively effected by the aforementioned
flawed ML practices. As a result, the predictive performance of these
models varies greatly with areas under the receiver operating
characteristic curve (AUROC) of 0.7-0.9.

\begin{itemize}
\tightlist
\item
  What we see in previous studies:

  \begin{itemize}
  \tightlist
  \item
    Differences in task definition (what is it that we want to predict?)
  \item
    Use of random forest only without discussion of interpretibility.
  \item
    Use of random forest without proper pipeline.
  \end{itemize}
\item
  What we do here:

  \begin{itemize}
  \tightlist
  \item
    To shed light on how much differences in modeling can affect the
    results, we performed an empirical analysis comparing several
    different modeling pipelines.
  \end{itemize}
\end{itemize}

\subparagraph{Summary of appraoch and
findings}\label{summary-of-appraoch-and-findings}

\begin{itemize}
\item
  Modeling pipelines were established for L2-regularized logistic
  regression, L1 and L2 support vector machines (SVM) with linear and
  radial basis function kernels, a decision tree, random forest and
  XGBoost.
\item
  These methods increase in complexity while decrease in
  interpretibility.
\item
  We established ML pipeline with held-out test data and performed 100
  data-splits to evaluate generalization and predicton performance of
  the ML method.
\item
  Applied to held-out test data, the mean AUROC varied from 0.68 (std ±
  0.04) to 0.82 (std ± 0.04).
\item
  Random Forest had the highest mean AUROC for detecting SRN and was
  less susceptible to overfitting compared to other methods.
\item
  Despite the lower mean AUROC value, the L1-regularized linear kernel
  SVM offered the greatest interpretability and stability.
\item
  In terms of computational efficiency, x trained the fastest, while y
  took the longest.
\item
  We found that cross-validation and testing AUROC could vary by as much
  as 0.06, highlighting the importance of a separate held-out test set
  for evaluation.
\item
  Aside from evaluating generalization and classification performance
  for each of these models, this study established standards for
  modeling pipelines of microbiome-associated machine learning models.
\end{itemize}

\subsection{Results}\label{results}

\subparagraph{AUROC results of 7 modeling approaches. (FIT +
OTUs)}\label{auroc-results-of-7-modeling-approaches.-fit-otus}

\subparagraph{Comparisons among the 7 modeling approaches. (FIT +
OTUs)}\label{comparisons-among-the-7-modeling-approaches.-fit-otus}

\begin{itemize}
\tightlist
\item
  Compare prediction performance, generalization performance and
  susceptibility to overfitting.
\end{itemize}

\subparagraph{Hyper-parameter tuning budgets and corresponding AUROC
values.}\label{hyper-parameter-tuning-budgets-and-corresponding-auroc-values.}

\begin{itemize}
\tightlist
\item
  This will show that we have used the right budgets to allow the model
  to pick the right hyper-parameter.
\end{itemize}

\subparagraph{AUC results of models with just
(FIT)}\label{auc-results-of-models-with-just-fit}

\begin{itemize}
\tightlist
\item
  This could be in supplemental. We want to make sure that using just
  FIT as a feature does worse than FIT+OTUs.
\end{itemize}

\subsection{Discussion}\label{discussion}

\subparagraph{Interpretation of modeling results in terms of
reproducibility, robustness, actionability, interpretibility and
susceptibility}\label{interpretation-of-modeling-results-in-terms-of-reproducibility-robustness-actionability-interpretibility-and-susceptibility}

\begin{itemize}
\item
  What are the metrics to talk about these concepts?
\item
  Emphasize trade-off results we observe in this study.
\end{itemize}

\subparagraph{Consideration of possible weaknesses for each
model}\label{consideration-of-possible-weaknesses-for-each-model}

\begin{itemize}
\tightlist
\item
  The interactions between the biomarkers may be nonlinear. Obviously,
  the linear models will not incorporate this because they are linear.
  Tools like linear models (e.g.~metastats, lefse, wilcoxon, etc) are
  likely worthless. How many OTUs would you find with these methods?
\end{itemize}

\subparagraph{Consideration of possible weaknesses for our approach and
chosen
dataset}\label{consideration-of-possible-weaknesses-for-our-approach-and-chosen-dataset}

\begin{itemize}
\tightlist
\item
  Why did we use only one dataset? Becuase we are comparing several
  methods on 1 dataset.
\end{itemize}

\subparagraph{Relationship of results to previous literature and broader
implications of this
work}\label{relationship-of-results-to-previous-literature-and-broader-implications-of-this-work}

\subparagraph{Prospects of future
progress}\label{prospects-of-future-progress}

\subsection{Methods}\label{methods}

\subparagraph{Brief explanation of study design/patient sampling and 16S
rRNA gene
sequencing/curation}\label{brief-explanation-of-study-designpatient-sampling-and-16s-rrna-gene-sequencingcuration}

\begin{itemize}
\tightlist
\item
  Baxter et al, 2016
\end{itemize}

\subparagraph{Analysis of data}\label{analysis-of-data}

\begin{itemize}
\item
  What are the features and what are the labels?

  \begin{enumerate}
  \def\labelenumi{\arabic{enumi}.}
  \item
    Features: Fecal hemoglobin concentration and 16S rRNA gene sequences
    from stool samples
  \item
    Labels: 490 patients as having advanced tumors (advanced adenoma or
    carinoma) or not (non-advanced adenoma or normal colon).
  \end{enumerate}
\item
  What is the data (temporal or not)? Assumptions we make when we use a
  dataset. What will the future data look like?
\item
  Pre-proccessing of the data
\item
  Machine Learning pipeline backbone. How do I split, train, validate
  and test?

  \begin{itemize}
  \tightlist
  \item
    A diagram to explain the modeling pipeline.
  \end{itemize}
\item
  Which methods are linear/non-linear? Talk about interpretibility.
\item
  Cross-validation and hyper-parameter tuning methods for each modeling
  method
\item
  Programming languages and packages/modules utilized
\item
  Statistical methods for comparison of model performance
\item
  Code/data availability
\end{itemize}


\end{document}
